### ISSUE TITLE
Closing client doesn't reject in-progress download

### ISSUE THREAD

**Issue by kami-** [patrickjuchli/basic-ftp#34]
**Describe the bug**
When a client is closed, the in-progress download's promise doesn't resolve or reject, so the promise just hangs.
Not sure if this is working as expected, but I think it would be better to have any in-progress download promise resolved or rejected when the client is cancelled.

**Example code**
```
const fs = require("fs");
const Client = require("basic-ftp").Client;

async function demo() {
    const host = "<host>";
    const user = "<user>";
    const password = "<password>";
    const localFilePath = "./test.txt";
    const remoteFilePath = "test-repo/@cba_a3/addons/cba_common.pbo";
    const client = new Client();
    client.ftp.verbose = true;
    console.log("Connecting to host...");
    await client.access({
        host: host,
        user: user,
        password: password,
    });
    console.log("Connected to host. Starting download...");
    client.download(fs.createWriteStream(localFilePath), remoteFilePath)
        .then(() => {
            console.log("Download finished.");
        })
        .catch(() => {
            console.log("Download failed.");
        });
    setTimeout(async () => {
        console.log("Closing FTP client...");
        await client.close();
        console.log("FTP client closed.");
    }, 200);
}

demo();
```

**Console output**
```
Connecting to host...
Connected to XXX.XXX.XXX.XXX:21
< 220 ProFTPD 1.3.5b Server (FTP server) [XXX.XXX.XXX.XXX]
Login security: No encryption
> USER XXX
< 331 Password required for XXX
> PASS ###
< 230 User XXX logged in
> TYPE I
< 200 Type set to I
> STRU F
< 200 Structure set to F
Connected to host. Starting download...
Trying to find optimal transfer strategy...
> EPSV
< 229 Entering Extended Passive Mode (|||56144|)
Optimal transfer strategy found.
> RETR test-repo/@cba_a3/addons/cba_common.pbo
< 150 Opening BINARY mode data connection for test-repo/@cba_a3/addons/cba_common.pbo (253334 bytes)
Downloading from XXX.XXX.XXX.XXX:56144 (No encryption)
Closing FTP client...
Closing connections.
FTP client closed.
```

**Which version of Node.js are you using?**
8.11.3

**Additional context**
My use case: I have a list of download promises that are combined into a single promise that sequentially download the files one after the other. Each download promise checks if the process has already been cancelled and will reject, otherwise starts the download normally. The problem is that an in-progress download won't resolve until it has finished or failed. This stops the whole promise chain from resolving. I thought that closing the client would reject any in-progress download promise, but they just hang.

**Comment by patrickjuchli** [comment]
This should be fixed in 2.16.1, let me know if this update works for you.

**Comment by kami-** [comment]
It works, the download promise gets rejected.
Thank you for the quick fix and for this awesome package!
